gateway/crates/federated-server/src/server.rs

```diff
@@ -34,14 +35,35 @@ use self::gateway::GatewayConfig;
 
 const DEFAULT_LISTEN_ADDRESS: SocketAddr = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 5000);
 
+/// Start parameter for the gateway.
+pub struct ServerConfig {
+    /// The GraphQL endpoint listen address.
+    pub listen_addr: Option<SocketAddr>,
+    /// The gateway configuration.
+    pub config: Config,
+    /// The config file path for hot reload.
+    pub config_path: Option<PathBuf>,
+    /// If true, watches changes to the config
+    /// and reloads _some_ of the things.
+    pub config_hot_reload: bool,
+    /// The way of loading the graph for the gateway.
+    pub fetch_method: GraphFetchMethod,
+    /// The opentelemetry tracer.
+    pub otel_tracing: Option<OtelTracing>,
+}
+
 /// Starts the self-hosted Grafbase gateway. If started with a schema path, will
 /// not connect our API for changes in the schema and if started without, we poll
 /// the schema registry every ten second for changes.
 pub async fn serve(
-    listen_addr: Option<SocketAddr>,
-    config: Config,
-    fetch_method: GraphFetchMethod,
-    otel_tracing: Option<OtelTracing>,
+    ServerConfig {
```

Comment:
:zap: **Suggestion** 
Ensure that the `config_hot_reload` functionality is properly validated and handled in the `serve` function to avoid unexpected behavior during runtime. Consider adding checks or logging to confirm that hot reload is functioning as intended.